src/core/StackExchange.Redis.Extensions.Core/Implementations/RedisDatabase.cs

```diff
@@ -58,6 +59,14 @@ public partial class RedisDatabase : IRedisDatabase
         /// <inheritdoc/>
         public ISerializer Serializer { get; }
 
+        /// <summary>
+        /// Utility method for creation of tags
+        /// </summary>
+        /// <param name="tag">The requested tag to set</param>
+        /// <param name="type">The type of the tagged object</param>
+        /// <returns>Tag string</returns>
+        public static string GenerateTagKey(string tag, Type type) => $"tag:{type.Name.GetHashCode()}:{tag}";
```

Comment:
GetHashCode is not an option, for the same type it will return each time different value

Reply:
should be then use the full name instead?

Reply:
I believe no, because in some time client can rename class or remove some property, but logic will work, because redis does not depends on model names, so we can just save it like separate key, and if tag will contain multy types models - it’s client problems :)

Reply:
It feels to me a little bit like offloading the heavy-lifting to someone else.
If you insist I'll remove it of course :)
The only reason I added the type name hash is to guard against multi types returned in one `GetByTag` request.

Reply:
for sure, but let take a look to ```GetAllAsync<IEnumerable<T>>(IEnumerable<string> keys)```
Implementation of this API does not guarantee that collection of keys that will be passed is related only to one type. So does it make sense to include this type check for tags? Also, the existing API does not depend on any type's name or hashcodes (which we can not use).

About ```GetByTag```. It can be implemented as separate methods for strings, hashes, and sets.  But we can have on that will do round trip within all tags.

Reply:
I see what you mean about no guaranties - you've got me convinced here.
I'll remove the type consideration